File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n=numbers.size();
       int l=0,r=n-1;
       while(l<r)
       {
            int cursum=numbers[l]+numbers[r];

            if(cursum>target)
            {
                r--;
            }
            else if(cursum<target)
            {
                l++;
            }
            else{
                return {l+1,r+1};
            }
       }
       return {};
    }
};

File: Data Structures & Algorithms/search-2d-matrix/submission-0.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int n=matrix.length;
        int m=matrix[0].length;

        int left=0;
        int right=m*n-1;

        while(left<=right)
        {
            int mid=(left+right)/2;

            int rpos=mid/m;
            int cpos=mid%m;

            if(matrix[rpos][cpos]==target)
            {
                return true;
            }
            else if(matrix[rpos][cpos]>target)
            {
                right=mid-1;
            }
            else{
                left=mid+1;
            }
        }
        return false;
    }
}
